### DISCUSSION TITLE
3章12節「HTTPサーバー」のコーディング

### DISCUSSION THREAD

**Discussion by satsukizzz** [satsukizzz/n-playspace-http#5] [Show and tell]
リポジトリ初めてのコーディングは私が行います！
プルリクエストを上げてみますので、皆さんレビューをお願いします 🙇 🙇

**Comment by yuukimaru130516** [comment]
よろしくお願いします！
ちなみに、Discussionsという機能があることを初めて知りましたｗ

**Comment by HappyKumamoto** [comment]
私もDiscussions、初めて知りました！
練習問題のプルリクを出すのがやっとのくせに、チームでの開発ってどうやるのか興味があるんです。
実際どんなふうにするのか教えていただきながら、少しでもお役に立てればと思います。
よろしくお願いします。m(_ _)m

**Comment by kenton116** [comment]
**初めまして！kenton116です！**
Discussions初めてです！
チームでの開発の練習になると思い、参加させていただきました！
どんなアプリにしますか？

**Reply by satsukizzz** [reply]
これ別トピックにしちゃいますne!

**Comment by kenton116** [comment]
僕もいっぱい協力します！

**Comment by risarisato** [comment]
初Discussions！
みなさんよろしくお願いします。

**Comment by HappyKumamoto** [comment]
みなさんよろしくお願いします！

リーダーすみません、クローンまで終わらせたところで、gitのブランチやマージが
理解できているか不安になりました。
第二章の復習をしてからレビューさせてください。

こういう話、ここに書いて良いのかどうかわかりませんが、よろしくお願いします。

**Reply by risarisato** [reply]
理解って・・・〇〇づけさんに参考して書いたのｗ
トレースしてると思考停止になりました笑

凄まじいプルリクにいいね！

DELETEとPUTを追加 

'use strict';
const http = require('http');

const fs = require('fs');           //ファイルシステムmodule読込
const fileName = './log.json';     //保存するファイル名宣言
//let log = [];
let log = new Map();

try {
   const akkData = fs.readFileSync(fileName, 'utf8');
   log = new Map(JASON.perse(allData));
}catch(ignore){
  console.log(fileName+'に、過去のログはありません');
}

const server = http
  .createServer((req, res) => {
    const now = new Date();

    let nichiji = {
      month:now.getMonth()+1,
      day:now.getDate(),
      year:now.getFullYear(),
      week:["日","月","火","水","木","金","土"][(now.getDay())],
      Hours:now.getHours(),
      Minutes:now.getMinutes(),
      Seconds:now.getSeconds()
    }

    let logData = `${nichiji.year}/${nichiji.month}/${nichiji.day}/(${nichiji.week})
    ${nichiji.Hours}:${nichiji.Minutes}:${nichiji.Seconds} \n
    「${req.socket.remoteAddress}」からアクセスがありました。\n`;

    log.set(now, logData);
    //saveLogs();
    saveTasks();
    console.info(logData);
    res.writeHead(200, {
      'Content-Type': 'text/plain; charset=utf-8'

    });

    switch (req.url){
       case '/log':
        res.write(logData);
       break;
       default:
       break;
    }

    switch (req.method) {
      case 'GET':
        res.write('GET ' + req.url + '\n');
        break;
      case 'POST':
        res.write('POST ' + req.url + '\n');
        let rawData = '';
        req
          .on('data', chunk => {
            rawData = rawData + chunk;
          })
          .on('end', () => {
            console.info('[' + now + '] Data posted: ' + rawData);
          });
        break;
      case 'DELETE ':
        res.write('DELETE' + req.url + '\n')
        break;
      default:
        break;
    }
    res.end();
  })
  .on('error', e => {
    console.error('[' + new Date() + '] Server Error', e);
    saveLogs();
  })
  .on('clientError', e => {
    console.error('[' + new Date() + '] Client Error', e);
    saveLogs();
  });

const port = 8000;
server.listen(port, () => {
  console.info('[' + new Date() + '] Listening on ' + port);
});

/**
 * ログをJSONファイルに保存する関数
 * JSON.stringify関数でJASONの文字列に変換して
 * 同期的にファイルに書き出す
 */
 function saveTasks(){
  fs.writeFileSync(fileName,JSON.stringify(Array.from(log)), 'utf8');
}

**Reply by risarisato** [reply]
って、突っ込みを入れましたが、私もみなさもgithubの使い方が不安です。
なわけで、試行錯誤して1度でいいからリーダにプルリクする！

OSSにプルリクしてmergeしていただいて人類に貢献するｗ

**Reply by HappyKumamoto** [reply]
おおー！
 とにかく１度でいいからプルリク！
了解です!
<(｀･ω･´)

**Reply by satsukizzz** [reply]
お！りさりさとさんの
https://github.com/satsukizzz/n-playspace-http/discussions/5#discussioncomment-1443457
これって、なんの機能でしょう。

よかったら、作りたい機能としてDiscussionsに新しくスレッド投稿してもらえませんか？？

**Reply by HappyKumamoto** [reply]
↑
あ、これは多分、私の3013プルリク
https://github.com/nnn-training/intro-curriculum-3013/pull/73
から、りさりさとさんがPUTを抜いてりさりさとさん流に書かれたコードだと思います。

私は昨年から受講していて、昨年は2019年のアーカイブで予習していたので、今年もついついプルリクも、過去のものを参考にさせています。

**Reply by HappyKumamoto** [reply]
↑
訂正(^^;
参考にさせてもらっています。

**Reply by HappyKumamoto** [reply]
たびたびすみません。

これ、同プルリクの、追加プルリクで、JSONファイルに保存の部分でした。
参考にさせていただいた2019プルリクから、自分なりに変えようとしてもうまくいかず、
変数名変えただけの状態だったので、Asazuke11よりと、メンション入れました。m(_ _)m

**Reply by satsukizzz** [reply]
なるほど～～
りさりさとさん結構いろいろ調べてますね 👀 
HappyKumamotoさんにも、よかったらプルリクエスト上げてもらいたいな 🤔

**Reply by HappyKumamoto** [reply]
そうそう、リーダーはじめ、りさりさとさんたちも、新たなお仲間も、
先生はもちろんだし、すごいかたばかりですよね！
のろのろですが、私も何かは上げたいと思います。

**Reply by risarisato** [reply]
いやいや、全然使い方がわかっていなくて💦
ただ、このスレッドはディスカッションをするという事ですね！
早く失敗して、同じ間違えを繰り返さないようにしていきます。

＞kumamotoさんのプルリクの実装がすごくて吹いたｗ
　それで、2章の復習ってオイって件ですｗ
　たのしくやりましょう。

**Reply by HappyKumamoto** [reply]
なるほど、そこへのツッコミでしたか〜
なんといっても、レジェンドのを真似ただけなので・・・ハハハ

>たのしくやりましょう
😀ですね！

**Comment by satsukizzz** [comment]
マージしました 🙇 
https://github.com/satsukizzz/n-playspace-http/pull/6
